son-gtklic/docs/swagger.yaml

```diff
@@ -0,0 +1,547 @@
+# this is an example of the Uber API
+# as a demonstration of an API spec in YAML
+swagger: '2.0'
+info:
+  title: Sonata's License Manager
+  description: Manage licenses Sonata services.
+  version: "1"
+# the domain of the service
+host: sonata-vm1.aws.atnog.av.it.pt
+# array of all schemes that your API supports
+schemes:
+  - http
+# will be prefixed to all paths
+basePath: /api/v1
```

Comment:
> ...this way, every version in the future would need to be implemented in the same file

Ouch! Really? Doesn't Python allow 'extensions'/'reopening' of classes?

As I've written this is still being designed. You're the first to implement it. This is not a peaceful issue, the majority of API providers include version in the URL. I disagree with this because I think the URL represents a resource, independent of the version of how to access it. But this is not the only possible reading of this. 

I think the trend nowadays is like GitHub does: `Accept`ing media headers like in `application/vnd.github.v3+json`. The idea is to maximize ease of use: if no version is indicated by the client of the API, then the most recent one is returned. If the version is indicated (possibly by an `Accept` header like GitHub's), then there must be a choice of which 'back-end' should answer. But there should always be a version returned. This is where the `X-Version` header enters. 

Suggestion: build a façade to any version built. Then you can split your implementation in several files...